src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/DefaultCostCalculator.java

```diff
@@ -131,9 +133,16 @@ public int calculateMinCost(int minTravelTime, int minNumTransfers) {
 
   @Override
   public int costEgress(RaptorTransfer egress) {
-    return egress.hasRides()
-      ? egress.generalizedCost() + transferCostOnly
-      : egress.generalizedCost();
+    if (egress.hasRides()) {
+      return egress.generalizedCost() + transferCostOnly;
+    } else {
+      // Remove cost that was added during alighting.
+      // We do not want to add this cost on last alighting since it should only be applied on transfers
+      // It has to be done here because during alighting we do not know yet if it will be
+      // a transfer or not.
+      var transferCost = stopTransferCost != null ? stopTransferCost[egress.stop()] : 0;
+      return egress.generalizedCost() - transferCost;
+    }
```

Comment:
```suggestion
    } else if(stopTransferCost != null) {
      // Remove cost that was added during alighting.
      // We do not want to add this cost on last alighting since it should only be applied on transfers
      // It has to be done here because during alighting we do not know yet if it will be
      // a transfer or not.
      return egress.generalizedCost() - stopTransferCost[egress.stop()];
    } else {
      return egress.generalizedCost();  
    }
```